crates/python/tests/test_isa.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+from os import path
+import json
+
+from qcs_sdk import get_isa
+from qcs_sdk.qpu.isa import InstructionSetArchitecture, Family
+
+
+
+def ignore_nones(value):
+    """Recursively ignore `None` values, useful for comparing json serializations."""
+    if isinstance(value, list):
+        return [ignore_nones(x) for x in value if x is not None]
+    elif isinstance(value, dict):
+        return {
+            key: ignore_nones(val)
+            for key, val in value.items()
+            if val is not None
+        }
+    else:
+        return value
+
+
+@pytest.fixture
+def aspen_m_3_json() -> str:
+    filepath = path.join(path.dirname(__file__), "fixtures/aspen-m-3.json" )
+    with open(filepath) as f:
+        contents = f.read()
+    return contents
+
+
+def test_isa_from_aspen_m_3_json(aspen_m_3_json: str):
+    isa = InstructionSetArchitecture.from_raw(aspen_m_3_json)
+
+    assert isa.architecture.family == Family.Aspen
+
+    node_ids = {node.node_id for node in isa.architecture.nodes}
+    assert len(node_ids) == 80
+
+    # TODO: example ISA has values that round when being converted to f32
+    # assert json.loads(isa.json()) == clean_nones(json.loads(aspen_m_3_json))
```

Comment:
Can these be changed to `f64`? https://docs.rs/qcs-api-client-openapi/latest/qcs_api_client_openapi/models/characteristic/struct.Characteristic.html

Reply:
We could try to use the `double` type in the schema and see if Rust picks that up as `f64`. I'm not seeing how best to specify that using `pydantic` -  there is the schema post-processing available in `Config`, but otherwise it seems like `float` types in Python end up as `float` in the schema which seems to be single-precision, and thus `f32`.

JSON numbers themselves are `f64`, so this change would make sense to me.

Reply:
Finally fixed this with https://crates.io/crates/qcs-api-client-openapi/0.5.2 dropping